content/en/docs/reference/access-authn-authz/certificate-signing-requests.md

```diff
@@ -0,0 +1,317 @@
+
+Once created, a CertificateSigningRequest must be approved before it can be signed.
+Depending on the signer requested, a CertificateSigningRequest may be auto-approved by a controller.
+Otherwise, a CertificateSigningRequest must be manually approved either via the REST API (or client-go)
+or via `kubectl certificate approve`. Likewise, a CertificateSigningRequest may also be denied, in which
+case the configured signer must not sign the request.
+
+Once approved, a signing controller must attempt to sign the request. A signer should apply its own policy
+restrictions, similar to the approval process, on the request to ensure it never signs an invalid request.
+There is currently not a mechanism for a signer implementation to report its inability to sign a request.
+Once the signer has determined that a request should be signed, it must sign the request and store the
+resulting certificate (and if appropriate, certificate chain) in the `status.certificate` field as PEM-encoded
+certificates.
+
+Once the `status.certificate` field has been populated, the request has been completed and clients can now
+fetch the signed certificate PEM data from the CertificateSigningRequest resource.
+
+In order to reduce the number of old CertificateSigningRequest resources left in a cluster, a garbage collection
+controller runs periodically. Depending on the state of the request, it will automatically delete requests older
+than a specified duration:
+
+* Approved requests: automatically deleted after 1 hour
+* Denied requests: automatically deleted after 1 hour
+* Pending requests: automatically deleted after 1 hour
+
+## Signers
+
+### Signers should define...
+
+All signers should provide information about how they work so that clients can predict what will happen to their CSRs.
+This includes:
+
+1. **Trust distribution**: how trust (CA bundles) are distributed.
+1. **Permitted subjects**: any restrictions on and behavior when a disallowed subject is requested.
+1. **Permitted x509 extensions**: including IP subjectAltNames, DNS subjectAltNames, Email subjectAltNames, URI subjectAltNames etc, and behavior when a disallowed extension is requested.
+1. **Permitted key usages / extended key usages**: any restrictions on and behavior when usages different than the signer-determined usages are specified in the CSR.
+1. **Expiration/certificate lifetime**: whether it is fixed by the signer, configurable by the admin, determined by the CSR object etc and behavior if an expiration different than the signer-determined expiration is specified in the CSR.
+1. **CA bit allowed/disallowed**: and behavior if a CSR contains a request a for a CA cert when the signer does not permit it.
+1. **optional**: Information about the meaning of additional CERTIFICATE PEM blocks in status.certificate, if different from the standard behavior of treating the additional certificates as intermediates, and presenting them in TLS handshakes.
```

Comment:
nit; this can wait for a follow-up PR:
```suggestion
1. **optional**: Information about the meaning of additional `CERTIFICATE PEM` blocks in `status.certificate`, if different from the standard behavior of treating the additional certificates as intermediates, and presenting them in TLS handshakes.
```